### PR TITLE
make id allow any kind of id and add smallid for the 32bit case

### PR DESCRIPTION
**Pull request by grosser**
@staugaard @apanzerj 

possible fix for https://github.com/zendesk/stronger_parameters/issues/7

### CODE REVIEW

File: lib/stronger_parameters/parameters.rb [hunk 1]
```diff
@@ -56,11 +56,11 @@ def integer64
...
 
       def id
-        integer & lte(2 ** 31) & gte(0)
+        integer & gte(0)
       end
 
-      def bigid
-        integer & lte(2 ** 63) & gte(0)
+      def smallid
+        integer & lte(2 ** 32) & gte(0)
```
Comment (ggrossman) [inline lib/stronger_parameters/parameters.rb]: for unsigned 32-bit integer: `lte(2 ** 32)` -> `lt(2 ** 32)`

Range is 0 to 4294967295 ((2^32) - 1)

Comment (grosser) [inline lib/stronger_parameters/parameters.rb]: done!

**Comment by grosser** [comment]
@staugaard what do you think ?
/cc @zendesk/octo

**Comment by apanzerj** [comment]
:+1:

**Comment by apanzerj** [comment]
cc @bquorning

**Comment by bquorning** [comment]
This project needs a changelog explaining why `Parameter.id` changes behavior between v1.0.3 and v2.0.0.

I actually think leaving `Parameter.id` at 32 bits would make sense for most applications. Doesn't Rails create 32-bit id columns by default?

:-1:

**Comment by grosser** [comment]
yeah kind of ... but not sure what we are protecting against, all an attacker could do is pass in high numbers that end up being saved as low numbers ...

**Comment by grosser** [comment]
yes 32bit is the default, but do we need to restrict users that choose to use unsigned / 64 bit / whatever size ? <-> what do we gain by doing that ?

**Comment by ggrossman** [comment]
The new behavior is an improvement ... I think it was a mistake to lock it to 32-bits in the initial implementation. Sure, the default may be `int(11)` in Rails, but that's just an initial/default setup. As we've seen, when a web app gets to a certain scale, 32-bit integers don't cut it anymore. Developers might even set up their initial schema with a bigger integer type in anticipation of future growth.

**Comment by bquorning** [comment]
In that case, I have misunderstood the purpose of this gem. I thought it was here to warn the user if he uses params of another type than the database – including max values for integers.

**Comment by grosser** [comment]
no, that is the purpose of this gem, this PR is just questioning if restricting id sizes makes much sense / causes more trouble then it helps

**Comment by grosser** [comment]
@staugaard what do you think ?

**Comment by staugaard** [comment]
:-1:

**Comment by grosser** [comment]
bigid it is ;)

**Comment by bquorning** [comment]
Yay

**Comment by apanzerj** [comment]
So this basically means we have to go through and replace id with bigid right?

**Comment by grosser** [comment]
yep ... good times ... or we can overwrite it with:

```Ruby
def id
  bigid
end
```
-> no bugs in the future + no giant commit

ideally something like `StrongerParameters.standard_id_size = :big`

**Comment by apanzerj** [comment]
monkey patch time

**Comment by apanzerj** [comment]
arg. from above:

yep ... good times ... or we can overwrite it with:

````
def id
  bigid
end
````
-> no bugs in the future + no giant commit

ideally something like `StrongerParameters.standard_id_size = :big`

**Pull request closed without merging**